django_elasticsearch_dsl/fields.py

```diff
@@ -56,7 +57,10 @@ def get_value_from_instance(self, instance, field_value_to_ignore=None):
             ):
                 try:
                     instance = getattr(instance, attr)
-                except (TypeError, AttributeError):
+                except (TypeError, AttributeError) as e:
+                    if isinstance(e, ObjectDoesNotExist):
+                        return None
+
```

Comment:
You could catch the `ObjectDoesNotExist` before the except block that catches `AttributeError`, instead of doing an `isinstance` check

Reply:
@stevelacey, you're right, that's much cleaner, not sure what I was thinking. I'll update the PR to use that style.